### PR TITLE
input latency integration

### PR DESCRIPTION
**Pull request by MikeLien**
@ShakoHo @askeing r?

### CODE REVIEW

File: lib/common/imageUtil.py [hunk 1]
```diff
@@ -246,8 +246,12 @@ def parallel_compare_image(input_sample_data, input_image_data, input_settings,
     image_fn_list.sort(key=CommonUtil.natural_keys)
 
     # generate search range
-    search_range = get_search_range(input_settings['exec_timestamp_list'], input_settings['default_fps'],
-                                    len(input_image_data))
+    if 'search_margin' in input_settings:
```
Comment (askeing) [inline lib/common/imageUtil.py]: I'll suggest refactoring like following code
```python
# TODO due to framework change still WIP, here's workaround
search_margin = input_settings.get('search_margin', 10)
search_range = get_search_range(input_settings['exec_timestamp_list'], input_settings['default_fps'],
                                len(input_image_data), search_margin)
```

File: lib/common/imageUtil.py [hunk 2]
```diff
@@ -267,7 +271,9 @@ def parallel_compare_image(input_sample_data, input_image_data, input_settings,
     for event_point in input_settings['event_points'][input_settings['search_direction']]:
         event_name = event_point['event']
         search_target = event_point['search_target']
-
+        shift_result_flag = False
+        if 'shift_result' in event_point:
+            shift_result_flag = event_point['shift_result']
```
Comment (askeing) [inline lib/common/imageUtil.py]: `shift_result_flag = event_point.get('shift_result', False)`

File: lib/helper/generatorHelper.py [hunk 3]
```diff
@@ -338,8 +349,8 @@ def calculate(env, crop_data=None, calc_si=0, waveform=0, revision="", pkg_platf
             current_time = time.time()
             elapsed_time = current_time - start_time
             logger.debug("Generate Video Elapsed: [%s]" % elapsed_time)
-            if waveform == 1:
-                output_waveform_info(calculator_result, env.waveform_fp, env.img_output_dp, env.video_output_fp)
+            # if waveform == 1:
+            #     output_waveform_info(calculator_result, env.waveform_fp, env.img_output_dp, env.video_output_fp)
```
Comment (askeing) [inline lib/helper/generatorHelper.py]: I will suggest remove the unused code :)

**Comment by askeing** [comment]
LGTM

**Comment by ypwalter** [comment]
currently, there are still some related issues:
1. need to merge with current master for not minimizing windows (however, it should be fine after merging to master)
2. console windows location and browser bar issues are required to be solved (I am on this for now)

please hold until further solving of these issues.

**Comment by ypwalter** [comment]
After locating the console window location to the right position, it is fine. r+
However, I have noted what parameters I used for this verification in #559.
Also, for windows, sikuli app foucs should also also be fixed for nightly so that console command window will be on the right position.

File: tests/regression/gdoc/test_chrome_gdoc_real_input_latency.sikuli/test_chrome_gdoc_real_input_latency.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+os.remove(sample1_fp)
+
+sleep(2)
+capture_width = int(sys.argv[6])
+capture_height = int(sys.argv[7])
+
+t1 = time.time()
+capimg1 = capture(0, 0, capture_width, capture_height)
+type('a')
+sleep(0.1)
```
Comment (ShakoHo) [inline tests/regression/gdoc/test_chrome_gdoc_real_input_latency.sikuli/test_chrome_gdoc_real_input_latency.py]: In normal condition, a should appear within 100ms, but if lag happened, that could lead the show up after 100 ms, and that will cause the calculation of AIL much smaller than expected.

**Comment by MikeLien** [comment]
#547

**Pull request merged**